zerver/actions/message_send.py

```diff
@@ -265,6 +283,26 @@ def notification_recipients(setting: str) -> Set[int]:
             # by default for new users.)
             wildcard_mention_user_ids = notification_recipients("wildcard_mentions_notify")
 
+        if possible_wildcard_mention and possible_topic_wildcard_mention:
+            # If the possibility exists for both the wildcard and topic wildcard mentions,
+            # we need to separately store the corresponding users to be notified.
+            #
+            # We can't give preference to wildcard mention and store
+            # 'wildcard_mention_user_ids' only, because after rendering it might be
+            # possible that there isn't any wildcard mention involved, only topic wildcard mention,
+            # e.g., @all in the code block and genuine @topic mention.
+            for user_id in wildcard_mention_user_ids:
+                if user_id in possibly_topic_wildcard_mentioned_user_ids:
+                    topic_wildcard_mention_user_ids.add(user_id)
+
+        if not possible_wildcard_mention and possible_topic_wildcard_mention:
+            possible_topic_wildcard_mention_user_ids = notification_recipients(
+                "wildcard_mentions_notify"
+            )
+            for user_id in possible_topic_wildcard_mention_user_ids:
+                if user_id in possibly_topic_wildcard_mentioned_user_ids:
+                    topic_wildcard_mention_user_ids.add(user_id)
```

Comment:
This logic is same as that of the one above, right? We should de-duplicate this and put it under a single `if possible_topic_wildcard_mention` conditional.

Reply:
See the [previous comment](https://github.com/zulip/zulip/pull/25828#discussion_r1216089745).